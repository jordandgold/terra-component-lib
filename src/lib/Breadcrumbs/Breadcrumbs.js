import React, { Component } from "react";
import PropTypes from "prop-types";

import "./Breadcrumbs.scss";

class Breadcrumbs extends Component {
  generateBreadcrumbs = () => {
    return this.props.breadcrumbs.map((breadcrumb, index) => {
      return (
        <a
          key={`breadcrumb-${index}`}
          className="breadcrumb"
          href={breadcrumb.link}
        >
          {breadcrumb.text}
        </a>
      );
    });
  };

  render() {
    return <div>{this.generateBreadcrumbs}</div>;
  }
}

export default Breadcrumbs;

Breadcrumbs.propTypes = {
  breadcrumbs: PropTypes.array.isRequired
};
