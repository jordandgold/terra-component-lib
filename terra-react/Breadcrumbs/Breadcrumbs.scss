@import "../global-styles/terra.scss";

.ter-breadcrumbs {
  &__list {
    @include border-radius(all, small);
    font-size: font-size("sm");
    padding: spacing("none");
    margin-top: 0;
    margin-bottom: spacing("xs");
    list-style: none;
  }
  &__list-item {
    display: inline-block;
    a {
    }
  }
  &__caret {
    margin-left: spacing("micro");
    margin-right: spacing("micro");
    fill: $breadcrumbs-caret-color;
  }
}
