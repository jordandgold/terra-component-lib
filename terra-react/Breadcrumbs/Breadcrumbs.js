import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import React, { Component } from "react";
import Icon from "../Icon/Icon";
import "./Breadcrumbs.scss";

var Breadcrumbs =
/*#__PURE__*/
function (_Component) {
  _inherits(Breadcrumbs, _Component);

  function Breadcrumbs() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, Breadcrumbs);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(Breadcrumbs)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _this.generateBreadcrumbs = function () {
      var breadcrumbArrow;
      var breadcrumbLink;
      return _this.props.breadcrumbs.map(function (breadcrumb, index) {
        if (_this.props.breadcrumbs.length - 1 === index) {
          breadcrumbArrow = null;
          breadcrumbLink = breadcrumb.text;
        } else {
          breadcrumbArrow = React.createElement(Icon, {
            className: "ter-breadcrumbs__caret",
            type: "open-caret-right-dark-8px",
            size: "8px"
          });
          breadcrumbLink = React.createElement("a", {
            href: breadcrumb.link
          }, breadcrumb.text);
        }

        return React.createElement("li", {
          className: "ter-breadcrumbs__list-item",
          key: "breadcrumb-".concat(index)
        }, breadcrumbLink, breadcrumbArrow);
      });
    };

    return _this;
  }

  _createClass(Breadcrumbs, [{
    key: "render",
    value: function render() {
      return React.createElement("nav", {
        className: "ter-breadcrumbs"
      }, React.createElement("ul", {
        className: "ter-breadcrumbs__list"
      }, this.generateBreadcrumbs()));
    }
  }]);

  return Breadcrumbs;
}(Component);

export default Breadcrumbs;