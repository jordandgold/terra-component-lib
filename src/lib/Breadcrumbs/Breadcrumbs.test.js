import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Breadcrumbs from "./Breadcrumbs";

describe("Breadcrumbs", () => {
  let wrapper;

  const mockProps = [
    { link: "https://www.google.com", text: "Google" },
    { link: "https://mail.google.com", text: "Gmail" }
  ];

  beforeEach(() => {
    wrapper = shallow(<Breadcrumbs breadcrumbs={mockProps} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("generateBreadcrumbs", () => {
    it("should return some jsx", () => {
      const expected = [
        '<li class="ter-breadcrumbs__list-item"><a href="https://www.google.com">Google</a><svg viewBox="0 0 8 8" class="ter-icon ter-icon--open-caret-right-dark-8px ter-icon--8px ter-breadcrumbs__caret"><use xlink:href="[object Object]#ter-icon--open-caret-right-dark-8px"></use></svg></li>',
        '<li class="ter-breadcrumbs__list-item">Gmail</li>'
      ];

      const result = wrapper
        .instance()
        .generateBreadcrumbs()
        .map(point => {
          return shallow(point).html();
        });

      expect(result).toEqual(expected);
    });
  });
});
