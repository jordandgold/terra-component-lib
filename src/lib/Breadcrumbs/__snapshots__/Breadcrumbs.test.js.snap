// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should match the snapshot 1`] = `
<div>
  <Component />
</div>
`;
